sw/device/lib/base/macros.h

```diff
@@ -98,4 +98,28 @@
  */
 #define OT_ATTR_WEAK __attribute__((weak))
 
+/**
+ * Spinwait with timeout in microseconds.
+ *
+ * It uses the Ibex's mcycle register value, with the knowledge of the CPU core
+ * clock frequency to estimate the elapsed time in microseconds. Caller must
+ * include the following header files:
```

Comment:
FYI: this kind of thing (caller must include these files) is a good sign that the macro should go in one of those files, or in another file that includes them.